cloudfoundry/provider/types_app.go

```diff
@@ -244,7 +259,20 @@ func (appType *AppType) mapAppTypeToValues(ctx context.Context) (*cfv3operation.
 	if !appType.Environment.IsNull() {
 		var env map[string]string
 		tempDiags = appType.Environment.ElementsAs(ctx, &env, false)
-		diags = append(diags, tempDiags...)
+		if tempDiags.HasError() {
```

Comment:
I do not understand the extensive refactoring of the code here that is not related to the intent of the PR focusing on lifecycle (or am I missing something).